### DISCUSSION TITLE
Welcome to ModernCppStarter Discussions!

### DISCUSSION THREAD

**Discussion by TheLartians** [TheLartians/Greeter#69] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by huand** [comment]
Hi, happy user of ModernCppStarter for my hobby projects to better structure my software architecture! thanks for making this package of best practices!

**Reply by TheLartians** [reply]
Hey, thanks for the feedback! I'm really happy this project is useful for you. :)

**Comment by ClausKlein** [comment]
Hi, great ideas, really!

Only one Question: why do you use so much `set_target_properties() `if there is a clean cmake command api?

```diff
+# WHY rename executable this way? CK
 set_target_properties(GreeterStandalone PROPERTIES CXX_STANDARD 17 OUTPUT_NAME "Greeter")
+# target_compile_features(GreeterStandalone PUBLIC cxx_std_17)
```

**Reply by TheLartians** [reply]
@ClausKlein thanks and welcome to the project! 👋 
About the api, I chose the option without too much research, could you perhaps elaborate on why `target_compile_features` is cleaner than `set_target_properties`.

**Reply by ClausKlein** [reply]
IMHO this [target_compile_features](https://cmake.org/cmake/help/latest/command/target_compile_features.html?highlight=target_compile_features) command is clearer to understand.

As same, i.e. nobody  should manipulate compiler options via `set(CMAKE_CXX_FLAGS …)`

Instead, we should use [target_compile_options](https://cmake.org/cmake/help/latest/command/target_compile_options.html?highlight=target_compile_options)
